diff --git a/node_modules/@dhis2/pwa/build/cjs/service-worker/set-up-service-worker.js b/node_modules/@dhis2/pwa/build/cjs/service-worker/set-up-service-worker.js
index ea02df0..1043224 100644
--- a/node_modules/@dhis2/pwa/build/cjs/service-worker/set-up-service-worker.js
+++ b/node_modules/@dhis2/pwa/build/cjs/service-worker/set-up-service-worker.js
@@ -58,7 +58,9 @@ function setUpServiceWorker() {
         url
       } = _ref;
       return url.endsWith('index.html');
-    });
+    }); // Make sure that this request doesn't redirect to a global shell
+
+    indexHtmlManifestEntry.url += '?redirect=false';
     (0, _workboxPrecaching.precache)([indexHtmlManifestEntry]); // Custom strategy for handling app navigation, specifically to allow
     // navigations to redirect to the login page while online if the
     // user is unauthenticated. Fixes showing the app shell login dialog
@@ -89,9 +91,10 @@ function setUpServiceWorker() {
 
 
       return true;
-    };
+    }; // Above, the index entry had the redirect param added:
+
 
-    const indexUrl = process.env.PUBLIC_URL + '/index.html';
+    const indexUrl = process.env.PUBLIC_URL + '/index.html?redirect=false';
 
     const navigationRouteHandler = _ref3 => {
       let {
diff --git a/node_modules/@dhis2/pwa/build/es/service-worker/set-up-service-worker.js b/node_modules/@dhis2/pwa/build/es/service-worker/set-up-service-worker.js
index 94d1d55..032a97f 100644
--- a/node_modules/@dhis2/pwa/build/es/service-worker/set-up-service-worker.js
+++ b/node_modules/@dhis2/pwa/build/es/service-worker/set-up-service-worker.js
@@ -44,7 +44,9 @@ export function setUpServiceWorker() {
         url
       } = _ref;
       return url.endsWith('index.html');
-    });
+    }); // Make sure that this request doesn't redirect to a global shell
+
+    indexHtmlManifestEntry.url += '?redirect=false';
     precache([indexHtmlManifestEntry]); // Custom strategy for handling app navigation, specifically to allow
     // navigations to redirect to the login page while online if the
     // user is unauthenticated. Fixes showing the app shell login dialog
@@ -75,9 +77,10 @@ export function setUpServiceWorker() {
 
 
       return true;
-    };
+    }; // Above, the index entry had the redirect param added:
+
 
-    const indexUrl = process.env.PUBLIC_URL + '/index.html';
+    const indexUrl = process.env.PUBLIC_URL + '/index.html?redirect=false';
 
     const navigationRouteHandler = _ref3 => {
       let {
